qiime2/core/type/collection.py

```diff
@@ -116,12 +120,14 @@ def is_element_expr(self, self_expr, value):
         """Since this is a dictionary, we often need to make sure to use its
         values not its keys.
         """
+        from qiime2.sdk import ResultCollection
+
         contained_expr = self_expr.fields[0]
-        if isinstance(value, self._view) and len(value) > 0:
+
+        if (isinstance(value, ResultCollection) or
+                isinstance(value, self._view)) and len(value) > 0:
```

Comment:
I'm not sure I have strong feelings either way about this. I did it this kinda way initially for obvious reasons (ResultCollection didn't exist when I started writing this), and for the other Collections _view is a Python built-in, honestly the ideal would kinda be for it to have dict and ResultCollection both as its views since it treats them identically. OR I guess in a perfect world, it would only ever see ResultCollection not dict, but given where this stuff gets called, that isn't feasible.